ta4j-core/src/main/java/org/ta4j/core/TradingRecord.java

```diff
@@ -158,6 +164,11 @@ default Position getLastPosition() {
      */
     Trade getLastTrade(TradeType tradeType);
 
+    /**
+     * @return the last trade recorded at the given time of 'index'
+     */
+    Trade getLastTradeRegardingIndex(int index);
```

Comment:
rename to

`Position getTrade(int index){..}`

The "regarding index" is already implicitly told by its needing parameter list `(int index)`